pkg/apis/v1alpha1/workload.go

```diff
@@ -34,11 +36,37 @@ const (
 	WorkloadLabelsMissingSupplyChainReason               = "WorkloadLabelsMissing"
 	NotFoundSupplyChainReadyReason                       = "SupplyChainNotFound"
 	MultipleMatchesSupplyChainReadyReason                = "MultipleSupplyChainMatches"
-	ServiceAccountErrorResourcesSubmittedReason          = "ServiceAccountError"
 	ServiceAccountSecretErrorResourcesSubmittedReason    = "ServiceAccountSecretError"
 	ResourceRealizerBuilderErrorResourcesSubmittedReason = "ResourceRealizerBuilderError"
+	ResolveTemplateOptionsErrorResourcesSubmittedReason  = "ResolveTemplateOptionsError"
+	TemplateOptionsMatchErrorResourcesSubmittedReason    = "TemplateOptionsMatchError"
 )
 
+// ValidWorkloadPaths Note: this needs to be updated anytime the spec changes
+var ValidWorkloadPaths = map[string]bool{
+	"workload.spec.source":                true,
```

Comment:
IIRC, in our templates it's valid to have both paths prefixed with or without `.`, like

```
    urlPath: .status.artifact.url
```

and

```
    urlPath: status.artifact.url
```

do you have any opinions on whether we should take that in consideration for these?

Reply:
Hmmmm, no real opinions. The code would work today except for the validations. Open to suggestions.